satpy/composites/__init__.py

```diff
@@ -1082,14 +1103,23 @@ def _get_and_sharpen_rgb_data_arrays_and_meta(self, datasets, optional_datasets)
             if 'rows_per_scan' in high_res.attrs:
                 new_attrs.setdefault('rows_per_scan', high_res.attrs['rows_per_scan'])
             new_attrs.setdefault('resolution', high_res.attrs['resolution'])
-            low_res_colors = ['red', 'green', 'blue']
-            low_resolution_index = low_res_colors.index(self.high_resolution_color)
+            colors = ['red', 'green', 'blue', None]
+            low_resolution_index = colors.index(self.high_resolution_color)
+            neutral_resolution_index = colors.index(self.neutral_resolution_color)
+            neutral_res = datasets[neutral_resolution_index] if self.neutral_resolution_color is not None else None
+
         else:
             LOG.debug("No sharpening band specified for ratio sharpening")
             high_res = None
+            neutral_res = None
             low_resolution_index = 0
+            neutral_resolution_index = 0
 
         if high_res is not None:
+            bands = locals()
+            colors = ['red', 'green', 'blue']
```

Comment:
Can we combine these in a more explicit dictionary? `bands = {"red": low_res_red, ...}`? Or actually, you never store anything except low res in it so could it just be `bands = {"red": low_res_red, "green": low_res_green, "blue": low_res_blue}` then you do the same overwrite for the high res color. If you did the high res assignment after the `ratio` is created, then the ratio `map_blocks` call could use `bands[self.high_resolution_color]` and `high_res` (decently clean...I think?).

Then...in the `with xr.set_options` block you could iterate over the colors (`for color in bands.keys()`) and do `if color != neutral_resolution_color and color != high_resolution_color: bands[color] = bands[color] * ratio`. Right? Thoughts? I think your idea to use dictionary-style access really makes this simple and then with my suggestions I think you can get rid of the `_index` and `neutral_res` variables.

Reply:
Now it's really cleaned up.